src/object/minishSizedEntrance.c

```diff
@@ -22,3 +23,18 @@ void sub_08090EC0(Entity* this) {
         LoadFixedGFX(this, 0x184);
     }
 }
+
+void sub_08090F00(Entity* this) {
+    if (this->type == 1) {
+        Entity* parent = this->parent;
+        u32 mask = 1 << this->field_0xf;
+        if (!(parent->field_0x20 & mask)) {
+            DeleteThisEntity();
+        }
+    }
+    if ((((gPlayerState.flags.all & 0x80) && sub_080041A0(this, &gPlayerEntity, 4, 4)) &&
+         (gPlayerEntity.height.HALF.HI == 0)) &&
+        (((u16)gPlayerState.field_0x90.HALF.LO) & gUnk_0812225C[this->type2])) {
```

Comment:
This has some extra parentheses, after removing those and running clang-format on this it should be
```suggestion
    if ((gPlayerState.flags.all & 0x80) && sub_080041A0(this, &gPlayerEntity, 4, 4) &&
        (gPlayerEntity.height.HALF.HI == 0) && (((u16)gPlayerState.field_0x90.HALF.LO) & gUnk_0812225C[this->type2])) {
```
making it a bit more clear there's just four clauses in a conjunction here.